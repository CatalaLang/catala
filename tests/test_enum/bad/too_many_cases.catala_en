## Article

```catala
declaration enumeration E:
  -- Case1 content integer
  -- Case2 content boolean
  -- Case3

declaration enumeration F:
  -- Case4

declaration scope A:
  context e content E
  context out content boolean

scope A:
  definition out equals match e with pattern
    -- Case1 of i : i = 0
    -- Case2 of b : b
    -- Case3 : false
    -- Case4 : true
```

```catala-test-inline
$ catala Interpret -s A
[ERROR]
This case matches a constructor of enumeration E but previous case were matching constructors of enumeration F

┌─⯈ tests/test_enum/bad/too_many_cases.catala_en:21.8-21.13:
└──┐
21 │     -- Case4 : true
   │        ‾‾‾‾‾
   └─ Article
#return code 123#
```
