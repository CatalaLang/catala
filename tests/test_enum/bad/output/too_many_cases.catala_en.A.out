[ERROR] This case matches a constructor of enumeration E but previous case were matching constructors of enumeration F
[ERROR] 
[ERROR]   --> test_enum/bad/too_many_cases.catala_en
[ERROR]    | 
[ERROR] 21 |     -- Case4 : true
[ERROR]    |        ^^^^^^^^^^^^
[ERROR]    + Article
